### PR TITLE
feat: 🎸 added dotlottie-rs integration

### PR DESCRIPTION
**Pull request by afsalz**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -78,30 +78,36 @@ dotLottieAnimationView.load(config)
...
 - `DotLottieAnimation.setSpeed(Float)` : Modifier the animation speed
 - `DotLottieAnimation.setLoop(Boolean)` : Make the animation to loop or not
 - `DotLottieAnimation.play()` : Play the animation
 - `DotLottieAnimation.pause()` : Pause the animation
 - `DotLottieAnimation.stop()` : Stop the animation
 - `DotLottieAnimation.load(Config)` : Setup the initial configuration
 - `DotLottieAnimation.setSegments(Float, Float)` : Defining the first and last frame
-- `DotLottieAnimation.setBackground(String)` : Set the animation background
+- `DotLottieAnimation.setPlayMode(Mode)` : Defining the first and last frame
+- `DotLottieAnimation.setBackground(Int)` : Set the animation background
```
Comment (theashraf) [inline README.md:106]:
```suggestion
- `DotLottieAnimation.setBackgroundColor(Int)` : Set the animation background
```

File: README.md [hunk 2]
```diff
@@ -78,30 +78,36 @@ dotLottieAnimationView.load(config)
...
 - `DotLottieAnimation.totalFrames` : Return the number of frame of the animations
 - `DotLottieAnimation.currentFrame` : Get the current frame
-- `DotLottieAnimation.mode` : Return the repeat mode of the animation, 
+- `DotLottieAnimation.mode` : Return the  playMode of the animation, 
 - `DotLottieAnimation.loop` : Return the repeat mode of the animation, 
-- `DotLottieAnimation.loopCount` : Return the number of loop of the animation, 
+- `DotLottieAnimation.loopCount` : Return the number of times animation has looped, 
 - `DotLottieAnimation.duration` : Get the animation duration in millis
 - `DotLottieAnimation.speed` : Get the animation speed
 - `DotLottieAnimation.autoPlay` : Get the animation speed
```
Comment (theashraf) [inline README.md:89]:
```suggestion
- `DotLottieAnimation.autoplay` : Get the animation speed
```

File: sample/src/main/res/layout/main.xml [hunk 3]
```diff
@@ -54,8 +54,14 @@
             <com.lottiefiles.dotlottie.core.widget.DotLottieAnimation
                 android:id="@+id/dot_lottie_view"
                 android:layout_width="200dp"
-                app:speed="1"
-                android:layout_height="200dp" />
+                android:layout_height="200dp"
+                app:dotLottie_autoPlay="true"
```
Comment (theashraf) [inline sample/src/main/res/layout/main.xml:58]:
```suggestion
                app:dotLottie_autoplay="true"
```

File: thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt [hunk 4]
```diff
@@ -48,12 +52,12 @@ class DotLottieAnimation @JvmOverloads constructor(
         get() = mLottieDrawable?.speed ?: error("DotLottieDrawable is null")
 
     val loop: Boolean
-        get() = loopCount != 1
+        get() = mLottieDrawable?.loop ?: false
 
     val direction: Int
```
Comment (theashraf) [inline thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt:57]: I thought we decided to remove the `direction` property, no ?

File: thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt [hunk 5]
```diff
@@ -134,19 +135,15 @@ class DotLottieAnimation @JvmOverloads constructor(
     }
 
     fun play() {
-        mLottieDrawable?.start()
-    }
-
-    fun setBackgroundColor(color: String) {
-        mLottieDrawable?.setBackgroundColor(color)
+        mLottieDrawable?.play()
     }
```
Comment (theashraf) [inline thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt:139]: setBackgroundColor is not longer in the dotlottie core kotlin bindings, but you can use `setConfig` to override the backgroundColor

File: thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt [hunk 6]
```diff
@@ -134,19 +135,15 @@ class DotLottieAnimation @JvmOverloads constructor(
...
 
     fun stop() {
         mLottieDrawable?.stop()
     }
 
-    fun setRepeatMode(repeatMode: Mode) {
-        mLottieDrawable?.setRepeatMode(repeatMode)
+    fun setPlayMode(repeatMode: Mode) {
+        mLottieDrawable?.setPlayMode(repeatMode)
     }
```
Comment (theashraf) [inline thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt:147]: I notice that sometimes the mode is called `playMode` and in other instances called `mode`, if we're going with the `playMode`, let's make it consistent

File: thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt (hunk 4 above)
Comment (afsalz) [reply thorvg-android/src/main/java/com/lottiefiles/dotlottie/core/widget/DotLottieAnimation.kt:57]: yeah true. ill remove.

**Review by theashraf — APPROVED** [review]

**Pull request merged**